library/core/src/option.rs

```diff
@@ -111,16 +111,20 @@
 //!
 //! ## Adapters for working with references
 //!
-//! * [`as_ref`] converts from `&Option<T>` to `Option<&T>`
-//! * [`as_mut`] converts from `&mut Option<T>` to `Option<&mut T>`
-//! * [`as_deref`] converts from `&Option<T>` to `Option<&T::Target>`
-//! * [`as_deref_mut`] converts from `&mut Option<T>` to
-//!   `Option<&mut T::Target>`
-//! * [`as_pin_ref`] converts from [`Pin`]`<&Option<T>>` to
-//!   `Option<`[`Pin`]`<&T>>`
-//! * [`as_pin_mut`] converts from [`Pin`]`<&mut Option<T>>` to
-//!   `Option<`[`Pin`]`<&mut T>>`
-//!
+//! * [`as_ref`] converts from <code>[&][][Option]\<T></code> to <code>[Option]<[&]T></code>
+//! * [`as_mut`] converts from <code>[&mut] [Option]\<T></code> to <code>[Option]<[&mut] T></code>
```

Comment:
I assume these are doing odd things to avoid linking `&` to Option - is there a reason you use `[&][]` for one and `[&mut] ` for the other?

Reply:
There’s no space after the `&`. The `[]` seems to be able to separate the two links without changing the fact that the `[&]: ../../std/primitive.reference.html "shared reference"` link is used. The space after `[&mut]` suffices to avoid the same problem there.

Reply:
Right, but why not use the space in both cases?

Reply:
Because we don’t want to render `& Option<T>` but `&Option<T>`

Reply:
It’s conventionally `&T` and `&mut T`, one has a space, the other doesn’t.

Reply:
Got it, that makes sense :+1: